kafka-rest/src/main/java/io/confluent/kafkarest/exceptions/DisabledOperationException.java

```diff
@@ -0,0 +1,13 @@
+package io.confluent.kafkarest.exceptions;
+
+import javax.ws.rs.core.Response.Status;
+
+public final class DisabledOperationException extends StatusCodeException {
+
+  public DisabledOperationException() {
+    super(
+        Status.METHOD_NOT_ALLOWED,
```

Comment:
I'm not sure 405 is the most appropriate response. The method(GET?) may be allowed but the resource is inaccessible. By saying the method isn't allowed it implies the resource is still valid you just need to use a different method. Perhaps 404 or even 410 would be more precise. It's a bit of a stretch but even 501 seems more appropriate.

Reply:
I think 404 for GET, and 405 for everything else is the most appropriate. That's what the documentation for 501 leads me to believe: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501

Reply:
That's fair I actually made this comment before I realized you were restricting resources as well as methods. 👍 In agreement.